src/vma/util/utils.h

```diff
@@ -161,24 +161,24 @@ int priv_read_file(const char *path, char *buf, size_t size, vlog_levels_t log_l
  * like above 'priv_read_file' however make sure that upon success the result in buf is a null terminated string
  */
 inline int priv_safe_read_file(const char *path, char *buf, size_t size, vlog_levels_t log_level = VLOG_ERROR){
-	int ret = priv_read_file(path, buf, size-1, log_level);
-	if (0 <= ret && size) buf[ret] = '\0';
+	int ret = -1;
+	if (size > 0) {
```

Comment:
different check in fact: size>0 and (size-1 >0) inside priv_read_file().
Will see how it can be reduced